src/execution/operator/helper/physical_limit.cpp

```diff
@@ -13,28 +15,61 @@ class PhysicalLimitOperatorState : public PhysicalOperatorState {
 	idx_t current_offset;
 };
 
+uint64_t get_delimiter(DataChunk &input, Expression *expr) {
+	DataChunk limit_chunk;
+	vector<LogicalType> types{expr->return_type};
+	limit_chunk.Initialize(types);
+	ExpressionExecutor limit_executor(expr);
+	if (expr->type == ExpressionType::BOUND_FUNCTION) {
+		limit_executor.Execute(limit_chunk);
+	} else {
+		limit_executor.Execute(input, limit_chunk);
+	}
+	auto limit_value = limit_chunk.GetValue(0, 0);
+	if ((limit_chunk.data[0].vector_type != VectorType::CONSTANT_VECTOR && limit_chunk.size() != 1) || limit_chunk.ColumnCount() != 1){
+		throw Exception("Value for Limit returns number of rows/columns different than 1");
+	}
+	bool success_cast = limit_value.TryCastAs(LogicalType::UBIGINT);
```

Comment:
This cast should not happen here. Instead, during binding the expected type of the expression should be specified. The system will then automatically add a cast operator if the type does not match. This is also used e.g. by the WHERE clause to signify that a boolean is expected. See `ExpressionBinder::target_type`.